fix_includes.py

```diff
@@ -1124,20 +1124,30 @@ def _DeleteDuplicateLines(file_lines, line_ranges):
   particular, it should not cover lines that may have C literal
   strings in them.
 
+  We only delete whole move_spans, not lines within them.
+
   Arguments:
     file_lines: an array of LineInfo objects.
     line_ranges: a list of [start_line, end_line) pairs.
   """
   seen_lines = set()
   for line_range in line_ranges:
     for line_number in range(*line_range):
-      if file_lines[line_number].type in (_BLANK_LINE_RE, _COMMENT_LINE_RE):
+      line_info = file_lines[line_number]
+      if line_info.type in (_BLANK_LINE_RE, _COMMENT_LINE_RE):
+        continue
+      if line_number != line_info.move_span[0]:
         continue
-      uncommented_line = _COMMENT_RE.sub('', file_lines[line_number].line)
-      if uncommented_line in seen_lines:
-        file_lines[line_number].deleted = True
-      elif not file_lines[line_number].deleted:
-        seen_lines.add(uncommented_line)
+      span_line_numbers = range(line_info.move_span[0], line_info.move_span[1])
+      line_infos_in_span = [file_lines[i] for i in span_line_numbers]
+      uncommented_lines = [
+              _COMMENT_RE.sub('', info.line) for info in line_infos_in_span]
+      uncommented_span = '\n'.join(uncommented_lines)
```

Comment:
It doesn't matter what the correct line ending for the file is; this joined string is only being used for comparison with other strings.  It just has to be consistent within this function.  It could be `'***'.join(...)` and it would work the same.

We don't even have to join the strings at all; could just store it as a tuple of lines.

Or, we could join with spaces, which would dedupe more cases, like this one:
```
template<typename T> class A;
template<typename T>
class A;
```
Any preference?

Reply:
Ah, I see. I never looked at the function as a whole, I thought `seen_lines` was returned. Cool! Yeah, let's use a space, then.

Reply:
Note that this logic may conflict with the changes in #643.  If I understand that correctly, when it is merged, this code will also need to strip line endings; should be a simple fix.